pkg/action/executor/machine_scaler_capi.go

```diff
@@ -41,19 +45,24 @@ type k8sClusterApi struct {
 	// Cluster API Resource client interfaces
 	machine    v1beta1.MachineInterface
 	machineSet v1beta1.MachineSetInterface
-
-	caGroupVersion string // clusterAPI group and version
 }
 
 // verifyClusterAPIEnabled Checks whether Cluster API is enabled.
-func (client *k8sClusterApi) verifyClusterAPIEnabled() error {
-	serviceString := fmt.Sprintf("ClusterAPI service \"%s\"", client.caGroupVersion)
-	_, err := client.discovery.ServerResourcesForGroupVersion(client.caGroupVersion)
+func (client *k8sClusterApi) verifyClusterAPIEnabled() bool {
+	return CheckClusterAPIEnabled(client.discovery)
+}
+
+func CheckClusterAPIEnabled(discoveryClient discovery.DiscoveryInterface) bool {
```

Comment:
Just a nit, I think the `CheckClusterAPIEnabled` is redundant, and we can rename `verifyClusterAPIEnabled` -> `isClusterAPIEnabled`.

Reply:
Thanks for the suggestion @ading1977. Updated. PTAL!